sdks/go/pkg/beam/core/runtime/graphx/coder.go

```diff
@@ -157,7 +157,7 @@ func (b *CoderUnmarshaller) makeCoder(c *pb.Coder) (*coder.Coder, error) {
 
 	case urnKVCoder:
 		if len(components) != 2 {
-			return nil, fmt.Errorf("bad pair: %v", c)
+			return nil, fmt.Errorf("could not make coder from proto %v, incorrect number of components for pair", *c)
```

Comment:
Further, take care when dereferencing pointers for use in printing: Some may have String() methods so you won't have to. See the following as an example: https://play.golang.org/p/WC2gf2wOBNf  
This applies everywhere, even to the protos, which typically have a String() method generated for their pointer types.

Reply:
That's really useful to know actually. I checked the existing String functions and removed all the dereferences I could (which was all of them).